airflow/jobs/base_job.py

```diff
@@ -308,3 +316,24 @@ def query(result, items):
             len(reset_tis), task_instance_str
         )
         return reset_tis
+
+    @provide_session
```

Comment:
By using this decorators it still uses a connection.

Using `with create_session` would be better

Reply:
Actually, on second though this is fine. The session is always passed in from the heartbeat, and there is already one there.